pkg/apis/internal.acorn.io/v1/unmarshal.go

```diff
@@ -135,22 +135,23 @@ func (in *Dependencies) UnmarshalJSON(data []byte) error {
 }
 
 func (in *Quantity) UnmarshalJSON(data []byte) error {
+	var (
+		s   string
+		err error
+	)
 	if !isString(data) {
-		var s int
-		if err := json.Unmarshal(data, &s); err != nil {
+		var d int
+		if err = json.Unmarshal(data, &d); err != nil {
 			return err
 		}
-		if s < 1000000 {
```

Comment:
This logic was moved to `ParseQuantity` to ensure this behavior is consistent each time a `Quantity` is parsed.